src/main/java/networkbook/model/person/Graduation.java

```diff
@@ -0,0 +1,235 @@
+        return match;
+    }
+
+    /**
+     * Parses user input {@code gradString} into an int representing the first calendar year of the academic year.
+     *
+     * @param gradString provided in user command
+     * @return corresponding first calendar year
+     */
+    public static int parseAcadYearStart(String gradString) {
+        String match = matchGroup(gradString, 1);
+        int shortYear = Integer.parseInt(match);
+        int addedPart = (shortYear < YEAR_THRESHOLD) ? YEAR_PART_HI : YEAR_PART_LO;
+        return Integer.parseInt(match) + addedPart;
+    }
+
+
+    /**
+     * Parses user input {@code gradString} into an int representing the second calendar year of the academic year.
+     *
+     * @param gradString provided in user command
+     * @return corresponding second calendar year
+     */
+    public static int parseAcadYearEnd(String gradString) {
+        String match = matchGroup(gradString, 2);
+        int shortYear = Integer.parseInt(match);
+        int addedPart = (shortYear < YEAR_THRESHOLD) ? YEAR_PART_HI : YEAR_PART_LO;
+        return Integer.parseInt(match) + addedPart;
+    }
+
+    /**
+     * Parses user input {@code gradString} into a {@code Semester}.
+     *
+     * @param gradString provided in user command
+     * @return corresponding semester
+     */
+    public static Semester parseSemester(String gradString) {
+        String match = matchGroup(gradString, 3);
+        if (match == null) {
+            return Semester.INVALID;
+        }
+        switch (match) {
+        case "1":
+            return Semester.S1;
+        case "2":
+            return Semester.S2;
+        default:
+            return Semester.INVALID;
+        }
+    }
+
+    /**
+     * Returns true if a given string representation of
+     * a graduation year and semester is valid.
+     */
+    public static boolean isValidGraduation(String gradString) {
+        String normalizedGradString = gradString.trim().toUpperCase();
+        if (normalizedGradString.matches(AY_REGEX)) {
+            int year1 = parseAcadYearStart(gradString);
+            int year2 = parseAcadYearEnd(gradString);
+            Semester sem = parseSemester(gradString);
+            return isValidGraduationYear(year1, year2) && isValidGraduationSemester(sem);
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Returns true if a pair of calendar years is valid (consecutive).
+     */
+    public static boolean isValidGraduationYear(int year1, int year2) {
+        return year2 - year1 == 1;
+    }
+
+    /**
+     * Returns true if a given graduation semester is valid.
+     */
+    public static boolean isValidGraduationSemester(Semester semester) {
+        requireNonNull(semester);
+        return semester != Semester.INVALID;
+    }
+
+    /**
+     * Returns human-readable representation of graduation year, e.g. "AY2022/2023 Semester 1".
+     */
+    public String getFullString() {
```

Comment:
You can add an assertion statement here to assert that the semester value is not `Semester.INVALID`. Same for the `toString` method below.